### ISSUE TITLE
Which hyperparameter setting to reproduce the SOTA result?

### ISSUE THREAD

**Issue by devnkong** [EricAlcaide/generalized-readout-phase#2]
Hello Eric, 

I'm trying to reproduce your results. I see in your paper the Softmax with p=1, beta=1 reaches the best. But when I tried to rerun the code you provided, the result is a little bit off. May I ask which file to use and if there are other settings?

Best

**Comment by hypnopump** [comment]
which results are you getting exactly? i could share the original file with the experiment results as logs

**Comment by devnkong** [comment]
By running https://github.com/EricAlcaide/generalized-readout-phase/blob/master/molhiv_softmax_good_beta_p1_beta1.ipynb
I got final result:

===========================
Final Test: 0.7864 ± 0.0135

**Comment by devnkong** [comment]
Hey Eric still cannot reproduce, am I using the right file? Thank you!

**Comment by hypnopump** [comment]
Gotcha. There are some differences between the files I uploaded here and the ones used for the experiment.

In short, stop trying to reproduce until I can have a look (probably later today or tomorrow).

I did some changes (basically strip cell outputs with logs and recopy the code from https://github.com/rusty1s/himp-gnn again to make sure that i removed all comments i had made).

The drop in performance is quite big actually. It seems something is wrong, will check and get back.

**Comment by hypnopump** [comment]
Okay, I've been inspecting the files this afternoon (the notebook with the original logs can be accessed here: https://drive.google.com/file/d/1IkZT6-mS-YZAgsv2t_qhAA-ZA3uK1uxb/view?usp=sharing ).

In short, I'm afraid **I did a catastrofic mistake** since I copied the experiment code from https://github.com/rusty1s/himp-gnn/blob/master/train_hiv.py instead of https://github.com/rusty1s/himp-gnn/blob/master/train_ogbhiv.py (in the file I copied for the original experiments, the splits don't follow the conventional splits for the OGB leaderboard). Then, when uploading the files to the repo, I cleaned the logs and copied again the code (this time I copied from the right file!) to remove the comments I had introduced in the original one.

**So, my results are not the true ones for the OGB benchmark (yours are).** I'm terribly sorry, this was by no means done in bad faith (the 2 files I got confused with are pretty similar, they just change the data splits!).

There are some things that, when seen in retrospection, where not fully coherent, but nearly everything seemed to make sense (information bottleneck, good consistent results, trainable params a bit different than the base ones, the run I did with the plain mean(·) readout yielded a worse result than the generalized ones, ...). 

Anyway, the experimental results are not what they were supposed to be since the experimental protocol is not the same as other entries in the leaderboard. Maybe generalized readout functions could still beat the HIMP baseline by adding more regularization to the model since my new experiments seem to indicate that the model might be overfitting the \beta and \p parameters (training performance is a bit better than the mean(·) baseline but that does not translate to val/test performance), but who knows (if you want to work on it, here's my email: ericalcaide1@gmail.com ). 

For now, I'm **taking down the prePrint** from Arxiv and **emailing to OGB team to update their leaderboard** so it does not mislead any more people.

**Comment by devnkong** [comment]
Hey Eric, thanks for your prompt response and detailed explanation. It's okay, this surely happens in research, and I definitely won't blame it so don't take it personally. Hope you all the best luck in the future research journey!

**Comment by sooheon** [comment]
Kudos for the open science. FWIW, I think you are on to something, I'm sure you've seen the DeeperGCNs paper which uses generalized fns in the same way as you, and there is also [PNA](https://arxiv.org/abs/2004.05718), which rather than using one generalized fn, just enumerates the different classes of desired fns and combines their results. I could see a hybrid approach, where you have multiple "heads" of generalized fns with their own p and beta.

**Comment by hypnopump** [comment]
Hi there! All ideas are welcome. 

Will try a prototype of tis asap (1-3 days) and provide feedback on the performance.

**Comment by weihua916** [comment]
@EricAlcaide It happens, no worries :) I removed your result from the leaderboard for now. Please feel free to make another submission once the result is fixed.

@devnkong Thank you for noticing the bug. Really appreciate your review effort!

Best,
Weihua -- OGB Team
